Mathlib/Analysis/CStarAlgebra/GNSConstruction/Defs.lean

```diff
@@ -0,0 +1,202 @@
+/-
+Copyright (c) 2025 Gregory Wickham. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gregory Wickham
+-/
+module
+
+public import Mathlib.Analysis.CStarAlgebra.GNSConstruction.PositiveLinearFunctional
+public import Mathlib.Analysis.InnerProductSpace.Completion
+
+/-!
+# Definitions of structures for the GNS (Gelfand-Naimark-Segal) construction
+
+In this file we do the "construction" part of the GNS construction. We define a Hilbert space
+as the completion of a quotient of A after we mod out by an appropriately constructed subspace.
+The inner product is induced by a fixed positive linear functional `f`.
+
+Most of the structures, theorems, and definitions in this file should not be referenced directly,
+but they are described below for the sake of clarity and completeness.
+
+## Main results
+
+- `f.GNS` : a type synonym of `A` that "forgets" the norm of `A` and bundles in a fixed
+  linear functional `f` so that we can construct an inner product and inner product-induced norm.
+- `f.GNS_Submodule`: the subspace of `A` defined as all elements `a : A` for which
+  `f (star a * a) = 0`.
+- `f.GNS_Quotient` : the quotient space of `(f.GNS) ⧸ (f.GNS_Submodule)`. We complete this space
+  with respect to its inner product-induced norm to produce `f.GNS_HilbertSpace`.
+- `f.GNS_Quotient_inner` : the inner product on `f.GNS_Quotient`.
+- `f.GNS_HilbertSpace` : the Hilbert space that we construct as the completion of `f.GNS_Quotient`.
+
+## References
+
+Most of this work follows from private course notes prepared by Professor Konrad Aguilar at Pomona
+College.
+
+For another, similar approach, see "A Primer on Spectral Theory" by Bernard Aupetit, the other main
+refence used here.
+-/
+
+@[expose] public section
+open scoped ComplexOrder
+--open PositiveLinearFunctional
+
+variable {A : Type*} [CStarAlgebra A] [PartialOrder A] (f : A →ₚ[ℂ] ℂ)
```

Comment:
GNS works just as well in non-unital algebras(†), so you probably don't want to restrict yourself to `CStarAlgebra`, but rather `NonUnitalCStarAlgebra`. Also, if you don't assume `StarOrderedRing A`, then the partial order on `A` is totally arbitrary, and not connected to the ⋆-algebra structure of `A`.

(†) Mathematically, this is true, but in Lean it's slightly complicated by the fact that we can't yet talk about left ideals in non-unital rings in Mathlib at the moment.